### PR TITLE
HDDS-4235. Ozone client FS path validation is not present in OFS.

### PR DESCRIPTION
**Pull request by ayushtkn**
## What changes were proposed in this pull request?

Added path name validation in OFS

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4235

## How was this patch tested?

Added UT

### CODE REVIEW

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java [hunk 1]
```diff
@@ -840,8 +842,12 @@ public String pathToKey(Path path) {
     if (!path.isAbsolute()) {
       path = new Path(workingDir, path);
     }
+    String key = path.toUri().getPath();
+    if (!OzoneFSUtils.isValidName(path.toUri().getPath())) {
```
Comment (adoroszlai) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java:846]: Avoid unnecessary duplicate conversion:

```suggestion
    if (!OzoneFSUtils.isValidName(key)) {
```

File: hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java [hunk 2]
```diff
@@ -817,7 +817,7 @@ public String pathToKey(Path path) {
     if (OzoneFSUtils.isValidName(key)) {
       key = path.toUri().getPath();
     } else {
```
Comment (adoroszlai) [inline hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java:819]: Not introduced in this patch, but would be nice to remove duplicate conversion:

```suggestion
    if (!OzoneFSUtils.isValidName(key)) {
```

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java [hunk 3]
```diff
@@ -1210,4 +1212,33 @@ public void testTrash() throws Exception {
...
+    checkInvalidPath(file1);
+  }
+
+  private void checkInvalidPath(Path path) throws Exception {
+    FSDataOutputStream outputStream = null;
+    try {
+      LambdaTestUtils.intercept(InvalidPathException.class, "Invalid path Name",
+          () -> fs.create(path, false));
+    } finally {
+      if (outputStream != null) {
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestRootedOzoneFileSystem.java:1238]: Seems like `outputStream` is never updated to non-`null` value, hence unnecessary.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @ayushtkn for taking care of this.

**Comment by ayushtkn** [comment]
Thanx @adoroszlai for the review, have addressed the above comments

**Review by adoroszlai — APPROVED** [review]
Thanks @ayushtkn for updating the patch.

**Comment by ayushtkn** [comment]
Thanx @adoroszlai for the review!!!

**Pull request merged**